src/components/Secrets/utils/secret-validation.ts

```
@@ -4,92 +4,106 @@ import * as yup from 'yup';
-        }),
-      ),
+export type SecretFormValidationOptions = {
+  isEditMode?: boolean;
+};
+
+const getSourceValidation = (isEditMode: boolean) =>
+  yup.object({
+    authType: yup.string(),
+    username: yup.string().when('authType', {
+      is: SourceSecretType.basic,
+      then: yup.string(),
```

Biome (2.4.4)

[error] Do not add then to an object.

(lint/suspicious/noThenProperty)